### DISCUSSION TITLE
Can you skip generation if there is an existing SRT?

### DISCUSSION THREAD

**Discussion by kuddukan1** [McCloudS/subgen#98] [General]
Like your example I am using aa.srt for the extension on the subgen files.  But if there is an eng.srt associated with a local media file, it will still create an aa.srt file too.  If there is an 'eng' internal subtitle then it correctly skips that one.  Is there something I should do to skip generation if there is an eng.srt subtitle file?

**Comment by McCloudS** [comment]
I typed out a giant thing saying why you could or couldn't, but I might be able to add it as an option.  What is your desired outcome/usecase?

Is the file name something like `Movie (2002).en.srt`?

Personally, I use a mix of the Plex webhook and Bazarr. I have aa still generate off of Plex webhooks and let Bazarr handle the subtitles as well (without using the Whisper Provider). That way I have the obvious AA showing (meaning Whisper generated them), and then eventually Bazarr will find a better non-AI generated subtitle and place it in the folder as eng. My player always defaults to EN subs, and if I'm not seeing any, I'll put on the AA subs.

**Comment by kuddukan1** [comment]
Ok I feel kind of stupid now LOL.  The scripts were looking for an aa.srt file since that's what I'm naming them.  I just changed the docker variable to create en.srt files so now it should not duplicate the thousands of en.srt files I already have as I run it over my local files.  I prefer not to use Bazarr, instead I am running subgen over my local folders.  I didn't make the connection that it would be looking for an aa.srt file just that that was the name of the new files it was creating.  Thank you.

**Reply by kuddukan1** [reply]
Previously, subgen was creating an aa.srt file alongside the existing en.srt files I already had.  My desired outcome was to only create an aa.srt file if there was no internal english subtitle OR an external en.srt file.  I can always just rename the new en.srt files after generation.

**Reply by McCloudS** [reply]
Oh okay, I made an edit to my above post that you may not have seen.  Changing that won't help because it'll still have the subgen name information in it (like filename.subgen.medium.aa.srt).

Your simplest option is to edit the script here: https://github.com/McCloudS/subgen/blob/620e97fb160bdfe0d227204ef3f199c189ad6d48/subgen.py#L110 to change it to something like... `subextension =  f".{namesublang}.srt"` and set NAMESUBLANG to en

This is probably a feature I need to add, I just need to think of how it could best be implemented.

**Comment by McCloudS** [comment]
Sorry for the back and forth.  I just added it as an option and the docker image is building now or you can use `UPDATE` to `True`.

`INFO:root:/movies/Maze Runner The Scorch Trials (2015)/Maze Runner The Scorch Trials (2015) Bluray-720p.mp4 already has an external en subtitle created for this, skipping it`

Essentially, set `NAMESUBLANG` to your desired language (`en`) and `SKIPIFEXTERNALSUB` to `True`.  I'm updating the documentation now.

**Comment by McCloudS** [comment]
Docker images are updated.  

See `SKIPIFEXTERNALSUB` in the readme, or:

`Skip subtitle generation if an external subtitle with the same language code as NAMESUBLANG is present. Used for the case of not regenerating subtitles if I already have Movie (2002).NAMESUBLANG.srt from a non-subgen source.`

**Comment by janusn** [comment]
Could it be simplified as setting a target language with ISO-639-2/T (or ISO_639_2_T as your document mentioned) and check for existing subtitles with either ISO-639-1 or ISO-939-2 in the name? Adding NAMESUBLANG and detected language to the mix if transcribe. E.g. `find . -type f -iregex "^${video-file-name-minus-extension}.*\.(${ISO_639_2_T}|${ISO_639_1}|${NAMESUBLANG}|${detected_lang})\.(srt|ass|ssa|smi|vtt)$"`

**Reply by McCloudS** [reply]
Most of the functionality was rewritten since this original discussion.

https://github.com/McCloudS/subgen/blob/5c84381ee99cd8123eea8233e579edb303d99f32/subgen.py#L1180 actually does this @ line 1221 by checking if existing external subs have a part that matches a language code via `if any(LanguageCode.from_string(part) == target_language for part in parts):`

**Comment by wm1801z** [comment]
I am working on creating an AI version for each of my files but cant seem to set the variables to ignore the current .srt files if present.
